### PR TITLE
feat: add basic support for esm

### PR DESCRIPTION
**Pull request by jeetiss**
hey @eunjae-lee! 
how are you? 🙌🏻

I mirgate same of my project to esm modules with `"type": "module"`, so now i can't use shipjs, because it load only `shipjs.config.js`
I added `cjs`, `mjs`, `js` possible extensions to config and use first existed, so it should solve the problem

also improve config for tests ❤️

### CODE REVIEW

File: packages/shipjs-lib/src/lib/config/loadConfig.js [hunk 1]
```diff
@@ -1,13 +1,30 @@
 import dotenv from 'dotenv';
 import { resolve } from 'path';
-import { existsSync } from 'fs';
+import { promises, constants } from 'fs';
```
Comment (Haroenv) [inline packages/shipjs-lib/src/lib/config/loadConfig.js:3]:
```suggestion
import { promises as fs, constants } from 'fs';
```

File: packages/shipjs-lib/src/lib/config/loadConfig.js [hunk 2]
```diff
@@ -1,13 +1,30 @@
...
 
 import defaultConfig from './defaultConfig';
 import mergeConfig from './mergeConfig';
 dotenv.config();
 
-export default function loadConfig(dir = '.', filename = 'ship.config.js') {
-  const fullPath = resolve(dir, filename);
-  const userConfig = existsSync(fullPath) ? require(fullPath) : {};
+const exist = path =>
+  promises
```
Comment (Haroenv) [inline packages/shipjs-lib/src/lib/config/loadConfig.js:10]:
```suggestion
  fs
```

File: packages/shipjs-lib/src/lib/config/loadConfig.js [hunk 3]
```diff
@@ -1,13 +1,30 @@
...
+    .access(path, constants.F_OK)
+    .then(() => path)
```
Comment (Haroenv) [inline packages/shipjs-lib/src/lib/config/loadConfig.js:12]:
```suggestion
    .then(() => true)
```

both are truthy, but I think exists should return a boolean

File: packages/shipjs-lib/src/lib/util/getAppName.js [hunk 4]
```diff
@@ -1,12 +1,14 @@
...
-export default function getAppName(dir = '.') {
-  const { appName } = loadConfig(dir);
+export default async function getAppName(dir = '.') {
+  const { appName } = await loadConfig(dir);
   if (appName) {
     return appName;
   }
-  const { name } = JSON.parse(readFileSync(resolve(dir, 'package.json')));
+  const { name } = JSON.parse(
+    await promises.readFile(resolve(dir, 'package.json'))
```
Comment (Haroenv) [inline packages/shipjs-lib/src/lib/util/getAppName.js:11]:
```suggestion
    await fs.readFile(resolve(dir, 'package.json'))
```

File: packages/shipjs-lib/src/lib/util/getAppName.js [hunk 5]
```diff
@@ -1,12 +1,14 @@
-import { readFileSync } from 'fs';
+import { promises } from 'fs';
```
Comment (Haroenv) [inline packages/shipjs-lib/src/lib/util/getAppName.js:1]:
```suggestion
import { promises as fs } from 'fs';
```

File: packages/shipjs/src/step/release/gatherRepoInfo.js [hunk 6]
```diff
@@ -7,9 +7,9 @@ import {
 } from 'shipjs-lib';
 import runStep from '../runStep';
 
-export default ({ remote, version, dir }) =>
-  runStep({ title: 'Gathering repository information.' }, () => {
-    const appName = getAppName(dir);
+export default async ({ remote, version, dir }) =>
```
Comment (Haroenv) [inline packages/shipjs/src/step/release/gatherRepoInfo.js:10]: is this first async/await needed?

File: packages/shipjs-lib/src/lib/config/loadConfig.js (hunk 3 above)
Comment (jeetiss) [reply packages/shipjs-lib/src/lib/config/loadConfig.js:12]: no, i use `path` in next operation chain

File: packages/shipjs/src/step/release/gatherRepoInfo.js (hunk 6 above)
Comment (jeetiss) [reply packages/shipjs/src/step/release/gatherRepoInfo.js:10]: yes

**Comment by jeetiss** [comment]
hello @Haroenv, thanks for review 🙌🏻

> since we process with babel, cjs & mjs should both work as expected.

Babel doesn't handle dynamically imported/required files, so it isn't working with with shipjs config file now.

**Comment by Haroenv** [comment]
CircleCI is down today I think

**Comment by jeetiss** [comment]
@Haroenv CircleCI isn't run test for PR from forks
Any chance that this PR will be released on this week?

**Comment by Haroenv** [comment]
I'll leave the merging to @eunjae-lee, he's back next week

**Review by eunjae-lee — APPROVED** [review]
Looks good. Thank you @jeetiss

**Comment by eunjae-lee** [comment]
@jeetiss released in 0.21.0 🎉

**Comment by jeetiss** [comment]
@eunjae-lee thanks 🙌🏻

**Pull request merged**